src/sentry/utils/performance_issues/detectors/sql_injection_detector.py

```diff
@@ -0,0 +1,143 @@
+from sentry.utils.performance_issues.types import Span
+
+MAX_EVIDENCE_VALUE_LENGTH = 10_000
+
+SQL_KEYWORDS = [
+    "SELECT",
+    "WHERE",
+    "AND",
+    "OR",
+    "NOT",
+    "IN",
+    "LIKE",
+    "LIMIT",
+    "ORDER BY",
+    "GROUP BY",
+    "HAVING",
+    "DISTINCT",
+    "JOIN",
+    "ON",
+    "AS",
+    "CASE",
+    "WHEN",
+    "THEN",
+    "ELSE",
+    "END",
+    "UNION",
+    "ALL",
+    "ANY",
+    "SOME",
+    "EXISTS",
+]
+
+
+class SQLInjectionDetector(PerformanceDetector):
+    __slots__ = "stored_problems"
+
+    type = DetectorType.SQL_INJECTION
+    settings_key = DetectorType.SQL_INJECTION
+
+    def __init__(self, settings: dict[DetectorType, Any], event: dict[str, Any]) -> None:
+        super().__init__(settings, event)
+
+        self.stored_problems = {}
+
+        query_string = event.get("request", {}).get("query_string", None)
+        if not query_string:
+            return
+
+        self.user_inputs = [
+            query_value[1] for query_value in query_string if self.keep_query_value(query_value)
+        ]
+
+    def visit_span(self, span: Span) -> None:
+        if not SQLInjectionDetector.is_span_eligible(span):
+            return
+
+        description = span.get("description")
+        fingerprint = fingerprint_span(span)
+        op = span.get("op", None)
+        spans_involved = [span["span_id"]]
+
+        if any(user_input in description for user_input in self.user_inputs):
+            for user_input in self.user_inputs:
+                fingerprint_description = span.get("description")
+                if user_input in fingerprint_description:
+                    fingerprint_description = fingerprint_description.replace(user_input, "")
+
+            self.stored_problems[fingerprint] = PerformanceProblem(
+                type=DBInjectionVulnerabilityGroupType,
+                fingerprint=self._fingerprint(fingerprint_description),
+                op=op,
+                desc=description[:MAX_EVIDENCE_VALUE_LENGTH],
+                cause_span_ids=[],
+                parent_span_ids=[],
+                offender_span_ids=spans_involved,
+                evidence_data={
+                    "op": op,
+                    "cause_span_ids": [],
+                    "parent_span_ids": [],
+                    "offender_span_ids": spans_involved,
+                    "transaction_name": self._event.get("description", ""),
+                },
+                evidence_display=[
+                    IssueEvidence(
+                        name="Offending Spans",
+                        value=get_notification_attachment_body(
+                            op,
+                            description,
+                        )[:MAX_EVIDENCE_VALUE_LENGTH],
+                        # Has to be marked important to be displayed in the notifications
+                        important=True,
+                    )
+                ],
```

Comment:
I'd try to look for other useful data to throw into the issue. `query_key` is one I pointed out before, but maybe a URL or other tags? Other elements of the `event.request` data maybe?